app/presenters/autocomplete_items_presenter.rb

```diff
@@ -23,15 +26,22 @@ def pages_with_autocomplete_component
   end
 
   def messages
-    @messages ||=
-      pages_with_autocomplete_component.each_with_object([]) do |page, arry|
-        page.components.each do |component|
-          next unless component.uuid.in?(component_uuids_without_items)
-
-          arry.push(
-            { component_title: component.humanised_title, page_uuid: page.uuid }
-          )
-        end
+    pages_with_autocomplete_component.each_with_object([]) do |page, arry|
+      page.components.each do |component|
+        next unless component.uuid.in?(component_uuids_without_items)
+
+        link = govuk_link_to(component.humanised_title, Rails.application.routes.url_helpers.edit_page_path(service.service_id, page.uuid))
```

Comment:
```suggestion

```
could we move this method to a private method? Then we can just call it in the `description` variable. Something like:

```
  def messages
    pages_with_autocomplete_component.each_with_object([]) do |page, arry|
      page.components.each do |component|
        next unless component.uuid.in?(component_uuids_without_items)

        description = I18n.t(
          "publish.autocomplete_items.#{deployment_environment}.message",
          title: link(component, page)
        )

        msg = description.html_safe
        arry.push(msg)
      end
    end
  end

  private

  def link(component, page)
    govuk_link_to(
      component.humanised_title,
      Rails.application.routes.url_helpers.edit_page_path(service.service_id, page.uuid)
    )
  end
```